### PR TITLE
Reduce GitLab CI pipeline depth

### PR DESCRIPTION
**Pull request by achilleas-k**
Previously, our top-level gitlab-ci.yml included a call to `./test/scripts/configure-generators` which was used to generate a dynamic pipeline with jobs that would generate manifests for each distro/arch/image-type configuration.  Then, each of those manifest generation stages in the dynamic pipeline would generate another level of dynamic pipelines based on the manifest changes.

This PR reduces the depth of dynamic pipelines by 1 (to 1).  The gitlab-ci.yml in the repository is replaced by the configuration that was previously dynamically generated at the first level, with the other previous top-level jobs added (`init`, `verify`, `finish`).  A CI job makes sure that the generated config is up to date.

This change has two benefits:
- It reduces the total CI runtime *in all cases*.  The `configure-generators` step would take 3-4 minutes and would run every time.  Now it's done offline and a local run takes a few seconds.  The runtime of the job in CI was mostly spent setting up the runner.
- It makes the CI pipelines in GitLab easier to navigate since there is now only *one* level of dynamic generation.

Includes #576

### CODE REVIEW

**Comment by bcl** [comment]
Source prep is hitting that btrfs dev problem I've seen mentioned somewhere else...

**Review by thozza — APPROVED** [review]
Makes sense, thanks!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ jobs:
 
   prepare:
     name: "🔍 Check source preparation"
```
Comment (thozza) [inline .github/workflows/tests.yml:140]: You probably need to install the build deps here, now that this step also generates the list of images which compiles `images`.

So probably `sudo apt install -y libgpgme-dev libbtrfs-dev libdevmapper-dev`

Comment (achilleas-k) [reply .github/workflows/tests.yml:140]: Oops, yes.  Absolutely right.

**Comment by achilleas-k** [comment]
> It would be great to not run Gitlab CI at all if `Check source preparation` step fails.

This is a good suggestion.  I wanted to get away from what we do in osbuild-composer which is gate gitlab on all the unit tests, which can take some time.  But the source-prep stage, *especially* now that it's generating the CI config, should be in the same workflow as the GitLab trigger now.  I'll move some stuff around.

**Comment by achilleas-k** [comment]
Done and done.

**Review by thozza — APPROVED** [review]
Thanks.

**Pull request merged**